### PR TITLE
Add KTlint rule to ban Kotlin test asserts

### PR DESCRIPTION
**Pull request by hunterwerlla**
- AssertJ only
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudformation/yaml/YamlCloudFormationTemplateTest.kt [hunk 1]
```diff
@@ -124,10 +122,10 @@ Parameters:
         runInEdtAndWait {
             assertThat(template.parameters().toList()).hasSize(2)
             val tableTag = template.parameters().firstOrNull { it.logicalName == "TableTag" }
-            assertNotNull(tableTag)
-            assertNull(tableTag.defaultValue())
-            assertNotNull(tableTag.description())
-            assertNull(tableTag.constraintDescription())
+            assertThat(tableTag).isNotNull
+            assertThat(tableTag!!.defaultValue()).isNull()
```
Comment (kiiadi) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudformation/yaml/YamlCloudFormationTemplateTest.kt:126]: yeah this is the only problem with using AssertJ is you don't get the smart-typing. I'd prefer these were re-written using `software.aws.toolkits.core.utils.test.notNull`

```
assertThat(tableTag).notNull.satisfies { tag ->
  assertThat(tag.defaultValue()).isNull()
  assertThat(tag.description()).isNotNull()
  assertThat(tag.constraintDescription()).isNull()
}
```

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunctionTest.kt [hunk 2]
```diff
@@ -53,36 +54,37 @@ Resources:
       Handler: helloworld.App::handleRequest
       Runtime: foo
       Timeout: 800
-""")
+"""
+        )
     }
 
-    @Test
+    @Test(expected = IllegalArgumentException::class)
```
Comment (kiiadi) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunctionTest.kt:61]: Any reason to use this vs `assertThatThrownBy`

Comment (hunterwerlla) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunctionTest.kt:61]: We seem to use expected a lot. Beyond that, not really. Will change to use `assertThatThrownBy`

Comment (hunterwerlla) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunctionTest.kt:61]: Maybe we should ban expected too :D

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudformation/yaml/YamlCloudFormationTemplateTest.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudformation/yaml/YamlCloudFormationTemplateTest.kt]: Sometimes this doesn't work (mostly due to tests being too complicated), so unbanned the kotlin test assertNotNull as it handles nullability properly

**Pull request merged**